### PR TITLE
Implement pinned.tab.frozen setting (issue #4400)

### LINKED ISSUE #4400: Add setting to allow navigation in pinned tabs
I should be able to open any url in a pinned tab as long as it's on the same domain.

### PR DESCRIPTION
**Pull request by beelzebielsk**
Implement a new setting, `pinned.tab.frozen` (boolean), which when false
allows a user to navigate to new URLs in a pinned tab (default behavior
is to have `pinned.tab.frozen = true`, in which no navigation is allowed).

closes #4400

Changes to be committed:
      modified:   qutebrowser/browser/browsertab.py
        Add `navigation_blocked` method to AbstractTab, which returns
        true if navigation is allowed on a tab.
      modified:   qutebrowser/browser/commands.py
        Change `openurl` to use `navigation_blocked` method of a tab to
        determine behavior while navigating a pinned tab.
      modified:   qutebrowser/components/misccommands.py
        Change `home` to use `navigation_blocked` method of a tab to
        determine behavior while navigating a pinned tab.
      modified:   qutebrowser/config/configdata.yml
        Add `pinned.tab.frozen` config.
      modified:   tests/end2end/features/tabs.feature
        Add tests for :open and :home on pinned tabs with
        `pinned.tab.frozen` set to false

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/4508)

### CODE REVIEW

**Comment by beelzebielsk** [comment]
In response to (from IRC):
> < The-Compiler> beelzebielsk: is it intended that you changed the indentation/logic in
:open command?

Sort of? I didn't want to write `curtab.load_url(cur_url)` in more than once place, and this was the way I saw to do it. If you don't want to change the indentation level from what it was before, then this ought to have the same behavior:

```
                if curtab is None:
                    if count is None:
                        # We want to open a URL in the current tab, but none
                        # exists yet.
                        self._tabbed_browser.tabopen(cur_url)
                    else:
                        # Explicit count with a tab that doesn't exist.
                        return
                elif curtab.navigation_blocked():
                    message.info("Tab is pinned!")
                else:
                    curtab.load_url(cur_url)
```

**Comment by jgkamat** [comment]
Adam Ibrahim writes:

>                 if curtab is None:
>                     if count is None:
>                         # We want to open a URL in the current tab, but none
>                         # exists yet.
>                         self._tabbed_browser.tabopen(cur_url)
>                     else:
>                         # Explicit count with a tab that doesn't exist.
>                         return
>                 elif curtab.navigation_blocked():
>                     message.info("Tab is pinned!")
>                 else:
>                     curtab.load_url(cur_url)

I think this way is a little cleaner, it mirrors the old code a little more
closely (and should do the same thing).

Otherwise, everything else looks good to me! Thanks :)

**Comment by beelzebielsk** [comment]
Okay. So, what exactly do I do? Make a new commit with the change? If so, is there anything I have to do with this PR after I make that new commit?

**Comment by jgkamat** [comment]
One more thing I missed, sorry.

since we already have a 'tabs.pinned' setting class, could you rename the
setting to 'tabs.pinned.frozen', and move the definition right below the
definition for it just under tabs.pinned.shrink?

Here's a patch you can apply if you're busy :)

http://paste.debian.net/plain/1058856

File: qutebrowser/config/configdata.yml [hunk 1]
```diff
@@ -1768,6 +1768,11 @@ tabs.pinned.shrink:
   type: Bool
   desc: Shrink pinned tabs down to their contents.
 
+tabs.pinned.frozen:
+    type: Bool
+    default: True
+    desc: If true, then pinned tabs cannot have their URL changed.
```
Comment (mschilli87) [inline qutebrowser/config/configdata.yml]: Could you re-write this to imperative mood for consistency with other options (see https://github.com/qutebrowser/qutebrowser/commit/494aceec451e307b0227b0fdd0f86d419dbc11d1)?
Maybe `Force pinned tabs to stay at a fixed URL.`?

Comment (beelzebielsk) [reply qutebrowser/config/configdata.yml]: Alright. Done.

**Comment by The-Compiler** [comment]
Seems okay from a very quick look, too busy with exams to take a proper one - @jgkamat feel free to merge if you feel confident enough.

Not sure what the error on Travis CI was, I triggered a rerun. Didn't seem related to this PR, but haven't seen it before...

**Comment by jgkamat** [comment]
I took a closer look at the failures and I'm pretty sure these two failures (in tab-give/take and undo) are on master as well, so someone will have to take a look at that seperately. I'm willing to support this change if it does cause problems though, so I'll merge it in now :)

Thanks for the contribution! :)

**Pull request merged**